### PR TITLE
WT-2985 checkpoint core dump

### PR DESCRIPTION
**Pull request by keithbostic**

### CODE REVIEW

**Comment by michaelcahill** [comment]
@keithbostic, sorry for the radio silence -- I've had back to back meetings for several days.

The latest version of this is what I was expecting: checkpoint should visit these pages, I think.  That test is intended to skip simple pages that were clean at the beginning of the checkpoint (or on disk) and become dirty while the checkpoint is running.  It should not be a common enough case for pages to be this state after a split has failed that there's any measurable performance penalty for visiting them.

File: src/btree/bt_sync.c [hunk 1]
```diff
@@ -9,6 +9,66 @@
...
+		    i = 0; i < mod->mod_multi_entries; ++multi, ++i)
+			if (multi->addr.addr == NULL)
+				return (false);
+
+	/*
+	 * Mark the tree dirty if skipping a page, the checkpoint marked it
+	 * clean and we can't skip future checkpoints until this page is
+	 * written.
+	 */
+	__wt_tree_modify_set(session);
```
Comment (michaelcahill) [inline src/btree/bt_sync.c]: Would you mind pull this up into the caller so the inlined function has no side effects?

**Comment by keithbostic** [comment]
@michaelcahill, this last set of changes is passing all of my tests, so if you're happy with it, it's ready to merge.

The reason I resisted the second approach was I don't like the checkpoint sync code looking inside the page modify structure, that's relatively ugly (not that the additional code in reconciliation was all that pretty, but at least it was in reconciliation).

Anyway, the only other idea I had was to add a new "multiblock with failed addresses" flag to the page modify structure, and that's not much better.

File: src/btree/bt_sync.c (hunk 1 above)
Comment (keithbostic) [inline src/btree/bt_sync.c]: Yes, you're right, that's better.

**Comment by michaelcahill** [comment]
Thanks @keithbostic, I'll merge.

**Pull request merged**